Add E2E Tests

sksaju opened the pull request:
<!--
Filling out this template is required.  Any PR that does not include enough information to be reviewed may be closed at a maintainers' discretion.  All new code requires documentation and tests to ensure against regressions.
-->

### Description of the Change
<!--
We must be able to understand the design of your change from this description.  The maintainer reviewing this PR may not have worked with this code recently, so please provide as much detail as possible.

Where possible, please also include:
- verification steps to ensure your change has the desired effects and has not introduced any regressions
- any benefits that will be realized
- any alternative implementations or possible drawbacks that you considered
- screenshots or screencasts
-->
Add E2E tests to the plugin.

<!-- Enter any applicable Issue number(s) here that will be closed/resolved by this PR. -->
Closes https://github.com/10up/embed-block-figma/issues/19

### How to test the Change
<!-- Please provide steps on how to test or validate that the change in this PR works as described. -->
Run E2E tests by running:

- npm install
- npm run cypress:run

### Changelog Entry

- Added - E2E tests

### Credits
<!-- Please list any and all contributors on this PR so that they can be added to this projects CREDITS.md file. -->
Props @sksaju, @jeffpaul @dkotter

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you are unsure about any of these, please ask for clarification.  We are here to help! -->
- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added [Critical Flows, Test Cases, and/or End-to-End Tests](https://10up.github.io/Open-Source-Best-Practices/testing/) to cover my change.
- [ ] All new and existing tests pass.

sksaju committed:
Add E2E Tests

sksaju committed:
fix js lint issue

sksaju committed:
fix js lint issue

dkotter commented on .github/workflows/cypress.yml:
```diff
@@ -0,0 +1,79 @@
+      matrix:
+        core:
+          - {name: 'WP latest', version: 'latest'}
+          - {name: 'WP minimum', version: 'WordPress/WordPress#5.7'}
```
Our minimum WP version for this plugin is 6.4
```suggestion
          - {name: 'WP minimum', version: 'WordPress/WordPress#6.4'}
```

dkotter commented on .github/workflows/cypress.yml:
```diff
@@ -0,0 +1,79 @@
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
```
```suggestion
      uses: actions/checkout@v4
```

dkotter commented on .github/workflows/cypress.yml:
```diff
@@ -0,0 +1,79 @@
+      working-directory: ${{ github.event.repository.name }}
+    - name: Cache node_modules
+      id: cache-node-modules
+      uses: actions/cache@v3
```
```suggestion
      uses: actions/cache@v4
```

dkotter commented on .github/workflows/cypress.yml:
```diff
@@ -0,0 +1,79 @@
+    - name: Display structure of downloaded files
+      run: ls -R
+      working-directory: ${{ github.event.repository.name }}
+    - name: Cache node_modules
```
Probably want a step before this where we install the proper node version, see https://github.com/10up/insecure-content-warning/blob/develop/.github/workflows/cypress.yml#L39

dkotter commented on .github/workflows/cypress.yml:
```diff
@@ -0,0 +1,79 @@
+          ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+    - name: Install dependencies
+      run: npm install
```
```suggestion
      run: npm ci
```

dkotter commented on .github/workflows/cypress.yml:
```diff
@@ -0,0 +1,79 @@
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+    - name: Install dependencies
+      run: npm install
+    - name: Composer
+      run: composer install
```
Not tested removing but don't think we actually need this step

dkotter commented on .gitignore:
```diff
@@ -34,3 +34,9 @@ Thumbs.db
 /db.php
+
+# Tests
+tests/cypress/downloads
```
Can probably just ignore the entire `tests` directory

jeffpaul replied:
updated in [9a33b0e](https://github.com/10up/embed-block-figma/pull/37/commits/9a33b0e856119bf1f774360dbe10d1c7961cceb7)

dkotter commented on package.json:
```diff
@@ -16,16 +16,31 @@
     "node": ">=18.0.0"
   },
+  "dependencies": {
+    "cypress-mochawesome-reporter": "^3.4.0"
```
I think this should be a dev-dependency

dkotter commented on tests/bin/set-wp-config.js:
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env node
```
I've not compared line by line but I know this is slightly different than the most recent version of this we've done. Can see that here: https://github.com/10up/insecure-content-warning/blob/develop/tests/bin/set-wp-config.js

jeffpaul replied:
Addressed in [4cb85c9](https://github.com/10up/embed-block-figma/pull/37/commits/4cb85c9b3d609202529cb39f4daaa003d30619d7)

dkotter commented on tests/cypress/e2e/admin.cy.js:
```diff
@@ -0,0 +1,16 @@
+		cy.login();
+	});
+
+	it('Open dashboard', () => {
+		cy.visit('/wp-admin');
+		cy.get('h1').should('contain', 'Dashboard');
+	});
```
We can probably remove this test

sksaju committed:
rename files

sksaju committed:
update cypress.yml

sksaju committed:
move cypress-mochawesome-reporter into dev dependancy

sksaju committed:
update tests/cypress/e2e/admin.cy.js

sksaju committed:
added Install node command in the cypress.yml

sksaju committed:
added cypress/plugins/index.js

sksaju committed:
Merge branch 'develop' into feature/19=e2e-tests

sksaju committed:
downgrade mochawesome-json-to-md package version

sksaju committed:
update cypress config

sksaju committed:
update cypress config

sksaju committed:
update set-wp-config.js

sksaju committed:
update set-wp-config.js

sksaju committed:
update set-wp-config.js

sksaju committed:
update set-wp-config.js

sksaju committed:
update cypress.yml workflow

sksaju committed:
update package-lock.json

sksaju committed:
update .wp-env.json and tests/bin/set-wp-config.js

sksaju committed:
update .github/workflows/cypress.yml

sksaju committed:
update .wp-env.json and tests/bin/initialize.sh

sksaju committed:
update tests/bin/initialize.sh

sksaju committed:
update e2e test file names

sksaju committed:
update embed-block-figma.test

sksaju committed:
fix js lint issue

sksaju committed:
update embed-block-figma.test

sksaju committed:
update embed-block-figma.test

sksaju committed:
update embed-block-figma.test

sksaju committed:
update embed-block-figma.test

sksaju committed:
update embed-block-figma.test

sksaju committed:
addes support command and update embed-block-figma.test.js

sksaju committed:
fix eslint issue

sksaju committed:
update embed-block-figma.test.js

sksaju committed:
update embed-block-figma.test.js

sksaju committed:
update embed-block-figma.test.js

sksaju committed:
update embed-block-figma.test.js

sksaju committed:
added setLocalStorage support command

sksaju committed:
update commands.js

sksaju committed:
update support command functions

sksaju committed:
update createPost support command

sksaju committed:
update e2e test case with  support commands

jeffpaul commented:
@dkotter any additional updates desired here before getting this merged in?

dkotter commented on package.json:
```diff
@@ -49,5 +61,8 @@
     }
+  },
+  "dependencies": {
+    "sharp": "^0.33.5"
```
Curious why we're adding a new dependency here? Doesn't seem like this is needed for this particular PR

jeffpaul replied:
done in [afdfc9d](https://github.com/10up/embed-block-figma/pull/37/commits/afdfc9dd846ee3c168b261c54acd0de64c4c59b2)

jeffpaul committed:
ignore entire tests directory

jeffpaul committed:
remove sharp dependency, update package-lock

jeffpaul committed:
Update set-wp-config.js

1. Added proper path handling using the path module
2. Added descriptive comments for each section
3. Improved error handling with specific error messages
4. Added proper exit codes (0 for success, 1 for errors)
5. Added success message when config is updated
6. Improved JSON formatting with indentation
7. Restructured the code to be more logical and readable
8. Added better error messages for both reading and writing operations

(thanks Cursor! ;)

jeffpaul commented:
I've noodled on this for a while today (with the help of Cursor) but cannot seem to get Cypress to properly trigger the block inserter or utilize a slash command to open the Figma block.  I'm going to put this down for now and come back later to see if a fresh approach shakes something loose.

dkotter committed:
Fix tests

dkotter commented:
E2E tests should be passing now in our minimum environment (6.5) and the current release (6.7). They are failing in 6.8-beta due to markup changes around the block inserter. This will be fixed in our utils package (see https://github.com/10up/cypress-wp-utils/pull/131) once a new release goes out there

dkotter committed:
Update Cypress WP Utils package

iamdharmesh committed:
Merge branch 'develop' into feature/19=e2e-tests

iamdharmesh reviewed (approved):
LGTM 🚀 . Thank you for your efforts here @sksaju @jeffpaul @dkotter.

iamdharmesh commented:
@dkotter I am unable to merge this because merging is blocked due to the requested changes from your end. Please merge it whenever you can. Thanks!
